docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
 Architecture Overview
 =====================
 
+Runhouse is a Python library that allows any application to flexibly and powerfully utilize remote compute
+infrastructure by deploying and calling remote services on the fly. It is principally designed for Machine
```

Comment:
Nit - I'm not in love with "on the fly" which connotes ad hoc. 
"through a simple, programmable API"? Or something else is fine too.